### ISSUE TITLE
NPM is unexpectedly taking up 10gb of storage on computer

### ISSUE THREAD

**Issue by Marak** [npm/npm#18680]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [x ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

NPM is unexpectedly taking up 10 gigabytes of storage on my computer.

#### How can the CLI team reproduce the problem?

Running this command in home directory:

```bash
find . -type d -name node_modules ! -wholename '*/node_modules/*' | xargs du -sch
```

### supporting information:

 - `npm -v` prints:  `5.4.2`
 - `node -v` prints: `v8.2.1`
 - `npm config get registry` prints: `https://registry.npmjs.org/`
 - Windows, OS X/macOS, or Linux?: `OS X/macOS`

**Comment by Klathmon** [comment]
Are you installing anything in particular that is causing this issue? Or is this just a complaint about data usage in general?

**Comment by ericmccarthy7** [comment]
You could always use something like [this](https://yarnpkg.com/en/docs/workspaces)

**Comment by ksmithut** [comment]
Steps to reproduce? Is this scanning every sub directory in your directory scanning every project you have on your computer with a node_modules directory?

**Comment by Marak** [comment]
Thanks @ericmccarthy7, workspaces in `yarn` seem like right idea.

Most of the space is due to various babel installs on projects each taking 600+mb each.

**Comment by Lindsor** [comment]
Yeah this is scanning all your modules in all your projects. Everytime you run npm install it downloads all the packages again. This is not an npm issue

**Comment by himanshub16** [comment]
In tried creating a react-native app. The `node_modules` directory took about 600 MB
